jhu-idc/idc-isle-dc
# Adds Google Tag Manager to IDC

Resolves https://github.com/jhu-idc/iDC-general/issues/373

This adds a [Google Tag Manager](https://www.drupal.org/project/google_tag) module to iDC.  **This does not track admin behavior, only end user pages**  This can be changed, but right now it only tracks end user actions. 

This also adds an environment variable in the `.env` file for the drupal container named: `DRUPAL_GTM_CONTAINER_ID`.   Right now the system will come up with no ID, and therefore no tracking. If an admin fills in the `.env` var above when starting the system, they will then begin tracking to that ID. 

The live configuration through the site will not show the  `DRUPAL_GTM_CONTAINER_ID` ID in its setup.  You can confirm it's using the correct one by looking at the page source for a user facing page. Setting the ID in drupal's UI will not work anymore, as it will be overridden by what's in the env variable.  

Note that the GTM module has containers -- we have one container for our ID.  Looking at the configuration for this  container in the GTM module will be confusing because the Container ID is not set in the active container for the site.   I created a container with and ID and then stripped out the ID in the configuration yml file. This was so there could be no active ID on the site.  If you edit the container in the drupal UI, it will force you to add a container ID - which may not work as it will be overridden by the environment variable.   I suggest only changing the ID via the environment variable in the `.env` file.

The GTM module will inject a script tag onto user facing pages. The script tag is to a file which contains the GTM snippet. 
 The Drupal GTM module creates this snippet and file.  This currently uses the public file system, as the private one didn't seem to work and the user was unable to download the google js file to their system.   The GTM js files can be found _inside_ the drupal container in the `/var/www/drupal/web/sites/default/files/google_tag/idc_gtm_info` directory. 
  
**To test:** 
 -  Start up the system (as is) and ensure that the script (via page source view) is not available in the source code on the sites home page.   
 -  Set the  `DRUPAL_GTM_CONTAINER_ID` in the `.env` file and `make reset`.  When it's done, ensure that there _is_ a GTM file link in the home page's source code (in the <head>), look at the file and verify that it contains the id you set in the `.env` file. Ensure that it is indeed tracking your use of the site by visiting the google analytics page for that ID you set.
